### ISSUE TITLE
SEC-1294: Support bean access in expression language

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-security#1540]
[Florent Ramiere](https://jira.spring.io/secure/ViewProfile.jspa?name=framiere) (Migrated from [SEC-1294](https://jira.spring.io/browse/SEC-1294?redirect=false)) said:

The expression language support in spring security is a great addition to the project.

Here is what we can currently do : 

@PreAuthorize("hasRole('ROLE_USER')")
public void basic()

@PreAuthorize("#age > 18")
public void usingArgs(int age)

Here is what I wish we could _also_ do :

@PreAuthorize("#age > #{mySpringBean.minAge}")
public void usingSpringBeansProperty(int age)

@PreAuthorize("#{mySpringBean.ageAuthorized(#age}}")
public void usingSpringBeansMethodAndArgs(int age)

Basically I wish we had the same functionnalities we have in spring core.
This is especially usefull for the Post

Currently I found the following way to use beans in the expression language 
- I access it via a custom method 
- that is defined in a custom MethodSecurityExpressionHandler 
- that overrides the "public EvaluationContext createEvaluationContext(Authentication auth, MethodInvocation mi)"
- then set the <expression-handler ref="" />
